cmd/kube-proxy/app/server_test.go

```diff
@@ -199,6 +200,12 @@ func Test_getProxyMode(t *testing.T) {
 		},
 	}
 	for i, c := range cases {
+		if c.expected == proxyModeIptables {
+			if _, err := os.Stat("/proc/sys/net/ipv4/conf/all/route_localnet"); os.IsNotExist(err) {
```

Comment:
I just want my unit tests to pass on Mac, which haven't passed for the last
5 months.  If you want me to refactor the proxier to accomplish that... :)
 Or an even simpler check for LookPath on "iptables".  Up to you.

On Wed, Jan 13, 2016 at 2:28 PM, Tim Hockin <notifications@github.com>
wrote:

> In cmd/kube-proxy/app/server_test.go
> <https://github.com/kubernetes/kubernetes/pull/19584#discussion_r49638001>
> :
>
> > @@ -199,6 +200,12 @@ func Test_getProxyMode(t *testing.T) {
> >  		},
> >  	}
> >  	for i, c := range cases {
> > +		if c.expected == proxyModeIptables {
> > +			if _, err := os.Stat("/proc/sys/net/ipv4/conf/all/route_localnet"); os.IsNotExist(err) {
>
> my point is mostly that we should inject interfaces that fake all of the
> things this decision-making function needs to make its decision.
> … <#1755025393_>
> On Wed, Jan 13, 2016 at 11:26 AM, Clayton Coleman <
> notifications@github.com> wrote: In cmd/kube-proxy/app/server_test.go <#19584
> (comment)
> <https://github.com/kubernetes/kubernetes/pull/19584#discussion_r49637496>>
> : > @@ -199,6 +200,12 @@ func Test_getProxyMode(t *testing.T) { > }, > } >
> for i, c := range cases { > + if c.expected == proxyModeIptables { > + if
> _, err := os.Stat("/proc/sys/net/ipv4/conf/all/route_localnet");
> os.IsNotExist(err) { FreeBSD has iptables too - in this case, the actual
> failure that happens first is this check. We could check for iptables but i
> don't know that our iptables proxier works on all iptables implementations.
> Either way … <#-509272425_> On Wed, Jan 13, 2016 at 2:21 PM, Tim Hockin <
> notifications@github.com> wrote: In cmd/kube-proxy/app/server_test.go <
> #19584 <https://github.com/kubernetes/kubernetes/pull/19584> (comment) <#19584
> (comment)
> <https://github.com/kubernetes/kubernetes/pull/19584#discussion_r49636851>>>
> : > @@ -199,6 +200,12 @@ func Test_getProxyMode(t *testing.T) { > }, > } >
> for i, c := range cases { > + if c.expected == proxyModeIptables { > + if
> _, err := os.Stat("/proc/sys/net/ipv4/conf/all/route_localnet");
> os.IsNotExist(err) { iptables alone is not enough? Because mac has iptables
> but not enough? The test should not even be hitting actual iptables. Maybe
> that is the real fix. … <#-610836060_> On Wed, Jan 13, 2016 at 10:02 AM,
> Clayton Coleman < notifications@github.com> wrote: In
> cmd/kube-proxy/app/server_test.go <#19584
> <https://github.com/kubernetes/kubernetes/pull/19584> <#19584
> <https://github.com/kubernetes/kubernetes/pull/19584>> (comment) <#19584
> <https://github.com/kubernetes/kubernetes/pull/19584> (comment) <#19584
> (comment)
> <https://github.com/kubernetes/kubernetes/pull/19584#discussion_r49625755>>>>
> : > @@ -199,6 +200,12 @@ func Test_getProxyMode(t *testing.T) { > }, > } >
> for i, c := range cases { > + if c.expected == proxyModeIptables { > + if
> _, err := os.Stat("/proc/sys/net/ipv4/conf/all/route_localnet");
> os.IsNotExist(err) { I don't know. Do you have a better one? iptables alone
> isn't enough. — Reply to this email directly or view it on GitHub <
> https://github.com/kubernetes/kubernetes/pull/19584/files#r49625755>. —
> Reply to this email directly or view it on GitHub <
> https://github.com/kubernetes/kubernetes/pull/19584/files#r49636851>. —
> Reply to this email directly or view it on GitHub <
> https://github.com/kubernetes/kubernetes/pull/19584/files#r49637496>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19584/files#r49638001>.
>